dali/operators/math/expressions/arithmetic_meta.h

```diff
@@ -359,13 +359,42 @@ REGISTER_UNARY_IMPL(ArithmeticOp::minus, -);
     static constexpr int num_outputs = 1;                                           \
   }
 
+// Specialization for mul and bool so we use && insteas of * so the compiler is happy
+#define REGISTER_BINARY_IMPL_BACKEND_MUL(DUMMY, BACKEND)                            \
+  template <>                                                                       \
+  struct arithm_meta<ArithmeticOp::mul, BACKEND> {                                  \
+    template <typename L, typename R>                                               \
+    using result_t = binary_result_t<L, R>;                                         \
+                                                                                    \
+    template <typename L, typename R>                                               \
+    DALI_HOST_DEV static constexpr result_t<L, R> impl(L l, R r) {                  \
+      auto l_ = static_cast<result_t<L, R>>(l);                                     \
+      auto r_ = static_cast<result_t<L, R>>(r);                                     \
+      return l_ * r_;                                                               \
+    }                                                                               \
+                                                                                    \
+    static inline std::string to_string() {                                         \
+      return "*";                                                                   \
+    }                                                                               \
+                                                                                    \
+    static constexpr int num_inputs = 2;                                            \
+    static constexpr int num_outputs = 1;                                           \
+  };                                                                                \
+                                                                                    \
+  template <>                                                                       \
+  DALI_HOST_DEV constexpr binary_result_t<bool, bool>                               \
```

Comment:
Wouldn't providing this be enough?
Also why add the `Dummy` parameter in the macro?

Reply:
...., right it should do.

Reply:
Done